### DISCUSSION TITLE
which image type of Depth should I choose of RGBD recording

### DISCUSSION THREAD

**Discussion by brz-hub** [microsoft/AirSim#4750] [Support Q&A]
Which data type is the depth data of rgbd camera, **DepthPlanar**, **DepthPerspective** or **DepthVis**? In ToF principle, I think we should choose DepthPerspective, but many people choose DepthVis or DepthPlanar. So which type should I choose to record data?

**Comment by CloudsRL** [comment]
Hello, i faced the same trouble, have you solve it? i need help too

**Reply by brz-hub** [reply]
You should select the DepthPlanar mode, where the distance from your target to the camera plane is also the z-value in the camera model. DepthVis is the visualization data that has been scaled by a percentage, not the actual distance, while in DepthPerspective, what you get is the distance from the target to the camera center, not the z-value. The TOF (Time of Flight) camera will internally convert the DepthPerspective obtained by the components into DepthPlanar.

**Reply by CloudsRL** [reply]
3ku so much, i will try it. I am tring to use AirSim to make dataset and use it in my SLAM program, but this problem confuse me very long time. I try to make depth into pointcloud and then turn it into Octomap, but i falled QAQ i hope this will help.

**Reply by brz-hub** [reply]
I hope you succeed in your program!    :)

**Reply by koyukan** [reply]
> 3ku so much, i will try it. I am tring to use AirSim to make dataset and use it in my SLAM program, but this problem confuse me very long time. I try to make depth into pointcloud and then turn it into Octomap, but i falled QAQ i hope this will help.

How was it? I am facing a similar challenge :D @CloudsRL

**Reply by CloudsRL** [reply]
thanks @brz-hub  , i have already finished it ^_^
DepthVis is not good, because it is for look.
DepthPerspective is ok, but i don't use it. Because when i use it, my uav feel the environment like a ball, so weird.
DepthPlanar is good, i use it. And by the way, i set the depth in pfm format, and then i use python to change it into the format i want. In this condition, u could setting the solution and distance as u like.(from my python, the longest distance is about 600m? but from my point, the RGB-D could only feel about 10-20m, so i think that's enough^_^)

**Reply by CloudsRL** [reply]
@koyukan i m  not good at GitHub, i hope u could see what i write down for you
